dao/src/main/java/com/epam/esm/dao/entity/audit/AuditListener.java

```diff
@@ -0,0 +1,36 @@
+package com.epam.esm.dao.entity.audit;
+
+import com.epam.esm.dao.entity.AuditHistory;
+import com.epam.esm.dao.entity.TableEntity;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PrePersist;
+import javax.persistence.PreRemove;
+import javax.persistence.PreUpdate;
+
+import static com.epam.esm.dao.entity.audit.Action.DELETED;
+import static com.epam.esm.dao.entity.audit.Action.INSERTED;
+import static com.epam.esm.dao.entity.audit.Action.UPDATED;
+
+public class AuditListener {
+
+  @PrePersist
+  public void prePersist(TableEntity tableEntity) {
+    perform(tableEntity, INSERTED);
+  }
+
+  @PreUpdate
+  public void preUpdate(TableEntity tableEntity) {
+    perform(tableEntity, UPDATED);
+  }
+
+  @PreRemove
+  public void preRemove(TableEntity tableEntity) {
+    perform(tableEntity, DELETED);
+  }
+
+  private void perform(TableEntity tableEntity, Action action) {
+    EntityManager entityManager = BeanUtil.getBean(EntityManager.class);
```

Comment:
@TimurRadko Could you inject `EntityManager` using `@Autowired`?

Reply:
As far as I can see there is no way to inject EntityManager into Listener through @Autowired.
https://stackoverflow.com/questions/22171221/how-to-inject-entitymanager-in-entitylisteners